### DISCUSSION TITLE
New mission filter - system region

### DISCUSSION THREAD

**Discussion by Zireael07** [endless-sky/endless-sky#7871] [Ideas]
For some of the ideas I've thrown around on Discord today, I heard that there is no way for the game (a mission) to do things when player arrives (and optionally spends a specified amount of time) to an area in a system. For a mission conclusion itself, I could do with source and destination being the same (and the NPC paying you upon landing back at the source planet) but still there's the need to detect if a player arrived to a certain spot.
I understand that the game has a fast polygon collision function, and if you've developed in any modern game engine, you're likely aware of the concept of _triggers_, also known as areas. This is unlikely to have any major impact on performance as we can short-circuit the check with an `if !player {return }` anyway

For the mission, I would like an `on trigger <time>` keyword so that I can show some text when the player reaches the target area and spends the time (likely not very long, I'm thinking 5 minutes, just something so that you have to actually spend some time there, not zip through it on your way to elsewhere). And I've already said the time can be optional, to make this even more universal.

The missions I was thinking about are intra-system ones, primarily in Sol, and the bodies I'm thinking of are NOT modeled in Endless Sky as landables (Trans-Neptunian Objects, Kuiper belt objects). The requirement to spend some time is to simulate 'exploring'/'scanning'/'assessing' (the TNOs) or actually landing and gathering some samples (Pluto&Charon). But I imagine this could be useful for the more usual kind of missions, too (e.g. someone at Sol sends you to 'see' any other place, maybe Alpha Centauri, maybe Betelgeuse, maybe some mostly fictional place)

**Comment by Zitchas** [comment]
I would definitely enjoy having that.

As far as that goes, I'd rather like to have proper timers, too.
